backend/internal/storage/postgres/schema/user.go

```diff
@@ -252,18 +252,27 @@ func (r *UserRepository) GetProfileByName(ctx context.Context, name string) ([]*
 
 	for profileRows.Next() {
 		var profile models.Profile
+		var profilePicture, bio sql.NullString
 		if err := profileRows.Scan(
 			&profile.UserID,
 			&profile.Username,
 			&profile.DisplayName,
-			&profile.ProfilePicture,
-			&profile.Bio,
+			&profilePicture,
+			&bio,
 			&profile.Followers,
 			&profile.Followed,
 		); err != nil {
 			return nil, err
 		}
 
+		if profilePicture.Valid {
```

Comment:
the changes in this file fix an issue where the backend wasn't returning profile photos properly. now it works (this endpoint powers profile searching)

![IMG_1445](https://github.com/user-attachments/assets/dc03d8b1-6281-4a09-a825-2c44d411601d)